### ISSUE TITLE
AES buils fails on VCS Compatability 

### ISSUE THREAD

**Issue by Jacob-Levy** [lowRISC/opentitan#13696]
When running CI(private) check with VCS, the AES fail duyring build:
```
ERROR: [Scheduler] [00:02:01]: [build]: [status] [aes:default: F]      // aes_masking
ERROR: [Scheduler] [00:04:16]: [build]: [status] [aes:default: F]      // aes_no_masking

```
@tjaychen @arnonsha  It is time to add VCS run/build to the release process...

**Comment by martin-lueker** [comment]
Hi @Jacob-Levy, 
   I can let @rasmus-madsen to look into the bugs here, as this is his block.
   AES is a block where the sign-off tool is Xcelium, as [you can see here]:(https://github.com/lowRISC/opentitan/blob/4a7932628d9bca84c7d631528df7175886e2f551/hw/ip/aes/dv/aes_base_sim_cfg.hjson#L15).  The CI(private) will fail when _smoke testing_ with VCS in the base configuration, though what is the policy for the rest of the tests?

**Comment by martin-lueker** [comment]
As a follow-up, I'm looking more closely into the AES configuration for CI(private) and I see that the masking and no-masking configurations are both being used.   However, with that said, I'm a bit confused why the current CI(private) wouldn't catch this.  What exactly are you trying to do?

**Comment by rasmus-madsen** [comment]
@Jacob-Levy @martin-lueker 
Thank you for bringing this to my attention - it seems  VCS is lacking some basic VIF support in this case.
@martin-lueker this is not caught because as you said Xcelium is the signoff tool - so everything in CI runs this with xcelium.

this issue will be fixed with the merge of #13645

**Comment by rasmus-madsen** [comment]
@Jacob-Levy VCS was never the default simulator for the AES block!
the issue is similar to what have been found before.
there are functionality that is supported in one tool but not in the other.

it is never caught by CI but always by the group using the other tool:
I think we have found about the same number of issues on each side.

having CI run both tools is too much overhead, which is why each ip has a signoff tool.

**Comment by arnonsha** [comment]
I am not completely sure  but it sounds like the private CI doesn't use the "default sign-off" tool for each block. Instead it uses only a single simulator. Maybe that should be fixed.

**Comment by rasmus-madsen** [comment]
@arnonsha it will use the simulator you tell it to i think.

Maybe if @Jacob-Levy give the command he uses to replicate this we will understand better!

**Comment by Jacob-Levy** [comment]
`util/dvsim/dvsim.py hw/top_earlgrey/dv/top_earlgrey_sim_cfgs.hjson --purge -w -i smoke --max-parallel 3 --tool=vcs --fixed-seed=1`

**Comment by rasmus-madsen** [comment]
@Jacob-Levy ahh - the dvsim does not look at what Tool is used for Signoff
it always assume VCS  unless you configure it differently,

to use xcelium add the `-t xcelium`

**Comment by martin-lueker** [comment]
Alright, my little test is complete.  I am able to run `dvsim` without the -t option, and it succeeds for all 43 test configurations.

```
          [   legend    ]: [Q: queued, D: dispatched, P: passed, F: failed, K: killed, T: total]                                                                                                                   
00:11:45  [    build    ]: [Q: 000, D: 000, P: 043, F: 000, K: 000, T: 043] 100%                                                                                                                                   
00:14:36  [     run     ]: [Q: 045, D: 016, P: 058, F: 000, K: 000, T: 119]  49%  i2c:0.i2c_host_smoke.1769218..   
```

I can also confirm [from the CI-private configuration](https://github.com/lowRISC/opentitan-private-ci/blob/0fb2f44c85c0f24bf997708fef6e45fa221a27c3/jobs.yml#L39) that it is _not_ using `-t` when running the smoke test.   So @Jacob-Levy, the `-t vcs` seems to be the source of your issue.

**Comment by Jacob-Levy** [comment]
The -t was delibrate since we are using VCS, but for the CI(private) run before doing a PR, I will remove it.
The big issue is that for the full-chip runs we are using VCS and if some Xeclium features end up there, we will fail to compile...

**Comment by tjaychen** [comment]
@sriyerg for more thoughts and details

i believe the direction we are going to move in for chip level is that both xcelium / vcs should run (xcelium doesn't yet, but that's something we are working on).

I would think long term block level really needs to be the same.. ie...we may choose to sign-off with a specific tool, but it would be great if multiple tools can run basic simulation (ignoring coverage related things for the moment).

**Comment by arnonsha** [comment]
I agree @tjaychen 
@martin-lueker, @rasmus-madsen  thank you for sorting this out.

**Comment by sriyerg** [comment]
V1 signoff mandates the smoke suite to pass with one alternate tool (VCS, if the signoff tool is Xcelium and vice versa). Since the AES block level is signed off with Xcelium, its smoke suite must pass with VCS at the least.

**Comment by rasmus-madsen** [comment]
@sriyerg all the tests pass with VCS - this was an issue with VCS and a newly created directed test

**Comment by sriyerg** [comment]
> 

Right, but @Jacob-Levy's first comment seems to indicate that the build is itself failing with VCS. I am able to reproduce it too:
```
Error-[NYI-CSTR-FTC5] NYI Constraint: function calls
../src/lowrisc_dv_aes_env_0.1/seq_lib/aes_control_fi_vseq.sv, 47
aes_env_pkg, "this.cfg.aes_control_fi_vif[this.if_num].get_if_size()"
  Calls of functions from virtual interfaces are not yet implemented in 
  constraints.
  Replace the function call with a temporary state variable that stores the 
  result of the function call. Assign the variable in pre_randomize().
```
This seems like an easy fix.
